spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/mapping/SpannerObjectMapperImpl.java

```diff
@@ -0,0 +1,360 @@
+		builder.put(Double.class, ValueBinder::toFloat64Array);
+		builder.put(Timestamp.class, ValueBinder::toTimestampArray);
+		builder.put(ByteArray.class, ValueBinder::toBytesArray);
+
+		iterablePropertyType2ToMethodMap = builder.build();
+	}
+
+	private static final Map<Class, BiFunction<Struct, String, List>>
+			readIterableMapping =
+			new ImmutableMap.Builder<Class, BiFunction<Struct, String, List>>()
+					.put(Boolean.class, AbstractStructReader::getBooleanList)
+					.put(Long.class, AbstractStructReader::getLongList)
+					.put(String.class, AbstractStructReader::getStringList)
+					.put(Double.class, AbstractStructReader::getDoubleList)
+					.put(Timestamp.class, AbstractStructReader::getTimestampList)
+					.put(Date.class, AbstractStructReader::getDateList)
+					.put(ByteArray.class, AbstractStructReader::getBytesList)
+					.build();
+
+	public SpannerObjectMapperImpl(SpannerMappingContext spannerMappingContext) {
+		Assert.notNull(spannerMappingContext,
+				"A valid mapping context for Spanner is required.");
+		this.spannerMappingContext = spannerMappingContext;
+	}
+
+	@Override
+	public <T> List<T> mapToList(ResultSet resultSet, Class<T> entityClass) {
+		ArrayList<T> result = new ArrayList<>();
+		while (resultSet.next()) {
+			result.add(read(entityClass, resultSet.getCurrentRowAsStruct()));
+		}
+		return result;
+	}
+
+	@Override
+	public <R> R read(Class<R> type, Struct source) {
+    R object = instantiate(type);
+		SpannerPersistentEntity<?> persistentEntity = this.spannerMappingContext
+				.getPersistentEntity(type);
+
+    PersistentPropertyAccessor accessor = persistentEntity.getPropertyAccessor(object);
+
+		for (Type.StructField structField : source.getType().getStructFields()) {
+			String columnName = structField.getName();
+			SpannerPersistentProperty property = persistentEntity
+					.getPersistentPropertyByColumnName(columnName);
+			if (property == null) {
+				throw new SpannerDataException(
+				    String.format("Error trying to read from Spanner struct column named %s"
+						+ " that does not correspond to any property in the entity type "
+                , columnName
+                , type));
+			}
+			Class propType = property.getType();
+			if (source.isNull(columnName)) {
+				continue;
+			}
+
+			boolean valueSet = false;
+
+			/*
+			 * Due to type erasure, binder methods for Iterable properties must be
+			 * manually specified. ByteArray must be excluded since it implements
+			 * Iterable, but is also explicitly supported by spanner.
+			 */
+			if (isIterableNonByteArrayType(propType)) {
+				valueSet = attemptReadIterableValue(property, source, columnName, accessor);
+			}
+			else {
+				Method getMethod = this.propertyReadMethodMapping.get(propType);
+				if (getMethod == null) {
+					getMethod = findReadMethod(source, propType);
+				}
+				if (getMethod != null) {
+					try {
+						accessor.setProperty(property, getMethod.invoke(source, columnName));
```

Comment:
We'll need to add checking for reading null columns (on this `getMethod.invoke`), as we have the [StructReader interface behave like  this](https://googlecloudplatform.github.io/google-cloud-java/0.25.0/apidocs/com/google/cloud/spanner/StructReader.html): 
"The getTypeName() methods should only be called for non-NULL values, otherwise a NullPointerException is raised; isNull(int)/isNull(String) can be used to test for NULL-ness if necessary."

Reply:
We have that right ? 

```
if (source.isNull(columnName)) {
			continue;
		}
```